PowerEditor/src/Notepad_plus.cpp

```diff
@@ -2571,8 +2571,12 @@ void Notepad_plus::addHotSpot()
 	_pEditView->execute(SCI_SETSTYLING, 0, 0);
 }
 
-bool Notepad_plus::isConditionExprLine(int lineNumber)
+bool Notepad_plus::isConditionExprLine(int lineNumber, LangType type)
 {
+	// These languages support single line blocks without braces.

```

Comment:
Do we really need that ?

Reply:
It explains why the function supports those languages, and why it does not support the other languages (Perl and Rust don't support single line blocks without braces). If someone adds new languages in the future, it should help them understand whether they should include the new language in that function or not.

Reply:
Was @donho talking about only the comment line, or the comment line plus the 3 lines that follow?  I find it hard to believe Don would object to the comment alone.

Reply:
Well the lines below are needed because Perl and Rust (and probably other languages I don't know about) _don't_ have single line control structures without braces, so it would be incorrect to indent there if the user did not insert a brace. (I assume this difference is the reason that Perl has not been supported for autoindent before, unlike every other braces language. Rust is pretty new so I sort of understand why it isn't supported yet.)

Reply:
@Kered13 I need a concrete example :
which code in Rust & Perl working and not working if we keep and remove these lines of code in PR ?

Reply:
With these lines, in Rust:
```
if condition
| <-- indentation goes here
```

In Perl:
```
if (condition)
| <-- indentation goes here
```


Without these lines, in Rust:
```
if condition
    | <-- indentation goes here
```

In Perl:
```
if (condition)
    | <-- indentation goes here
```

The problem with the second examples is that Rust and Perl _require_ braces in an if statement (and all similar statements). It is a syntax error to not have braces. Therefore this line should be treated as a simple multi-line statement, and Notepad++ does not automatically indent multi-line statements.

Reply:
In this case, I prefer add the follow lines:
```
			else if (type == L_PERL || type == L_RUST)
			{
				_pEditView->setLineIndent(curLine, indentAmountPrevLine);
			}
```
After
```
			else if (nextChar == '{')
			{
				_pEditView->setLineIndent(curLine, indentAmountPrevLine);
			}
```
instead of modifying the behaviour of `isConditionExprLine` function.
If makes the code simpler and more readable.